packages/devextreme/js/__internal/core/license/license_validation.ts

```diff
@@ -139,13 +158,63 @@ function getLicenseCheckParams({ licenseKey, version }: {
   }
 }
 
+export function showTrialPanel(buyNowUrl: string, version: string): void {
+  if (typeof customElements === 'undefined') {
+    return;
+  }
+
+  registerTrialPanelComponents();
+
+  const trialPanelTrigger = document.createElement(DX_LICENSE_TRIGGER_NAME);
+
+  trialPanelTrigger.setAttribute(trialPanelAttributeNames.buyNow, buyNowUrl);
+  trialPanelTrigger.setAttribute(trialPanelAttributeNames.version, version);
+
+  document.body.appendChild(trialPanelTrigger);
+}
+
+function shouldShowTrialPanel(
+  { preview, internal, error }: LicenseCheckParams,
+  licenseKey: string,
+  version: string,
+): boolean {
+  if (!error || internal) {
+    return false;
+  }
+
+  if (preview) {
+    const { major, minor, patch } = parseVersion(version);
+
+    const previousMajor = (
+      minor === 1 ? [major - 1, MAX_MINOR_VERSION, patch] : [major, minor - 1, patch]
+    ).join(SPLITTER);
```

Comment:
This can be extracted as `getPreviousMajorVersion`

Reply:
Probably `parseVersion` `getPreviousMajorVersion` and `ParsedVersion` (better simply `Version`) should be moved to a separate module (like 'core/utils/version')